src/object/cli_ec.c

```diff
@@ -2390,3 +2422,100 @@ obj_ec_tgt_oiod_init(struct obj_io_desc *r_oiods, uint32_t iod_nr,
 
 	return tgt_oiods;
 }
+
+/* Object EC encoding Helper */
+pthread_spinlock_t	oeh_lock;
+struct oeh_helper	oeh;
+
+static void *
+oeh_thread(void *arg)
+{
+	struct obj_reasb_req	*reasb_req;
+	struct oeh_work		*work;
+	int			 rc;
+
+	while (1) {
+		D_SPIN_LOCK(&oeh_lock);
+		if (!d_list_empty(&oeh.oeh_work_list)) {
+			work = d_list_entry(oeh.oeh_work_list.next,
+				struct oeh_work, oeh_link);
+			D_ASSERT(work != NULL);
+			d_list_del_init(&work->oeh_link);
+		} else {
+			work = NULL;
+		}
+		D_SPIN_UNLOCK(&oeh_lock);
```

Comment:
I think this is going to hurt performance, we cannot use spinlock in this kind of use case, I'd suggest to use mutex+cond, and put helper threads in sleep if there is nothing to do